streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDbIterator.java

```diff
@@ -81,7 +81,19 @@ public synchronized void close() {
 
     @Override
     public Bytes peekNextKey() {
-        if (!hasNext()) {
+        //TODO - bellemare - if I call hasNext() directly, I end up with stack overflow:
+        /*
+        <clip>
+            at org.apache.kafka.streams.state.internals.RocksDBPrefixIterator.hasNext(RocksDBPrefixIterator.java:45)
+            at org.apache.kafka.streams.state.internals.RocksDbIterator.peekNextKey(RocksDbIterator.java:84)
+            at org.apache.kafka.streams.state.internals.RocksDBPrefixIterator.hasNext(RocksDBPrefixIterator.java:45)
+            at org.apache.kafka.streams.state.internals.RocksDbIterator.peekNextKey(RocksDbIterator.java:84)
+        </clip>
+        */
+        if (!open) {
+            throw new InvalidStateStoreException(String.format("RocksDB iterator for store %s has closed", storeName));
+        }
+        if (!super.hasNext()) {
```

Comment:
This is a weird one. I end up with a stack overflow error as the above function calls just go back and forth. It seems like it's getting some weird scope problem. This works around it, but I am not sure if I stumbled on an existing bug or if it's something I introduced. I'm just not really sure how what I did could have caused this.

Reply:
This is due to the subclassing. If you instead wrap and delegate, this won't be a problem.